calabash/Classes/LPI.mm

```diff
@@ -0,0 +1,122 @@
+
+//
+//  LPI.mm
+//
+//  Created by Karl Krukow on 18/08/11.
+//  Copyright (c) 2011 LessPainful. All rights reserved.
+//
+
+#import "CalabashServer.h"
+#import "LPGitVersionDefines.h"
+
+#ifdef LP_SERVER_ID_KEY_VALUE
+static const char* kLPServerIdentifierKeyValue = LP_SERVER_ID_KEY_VALUE;
+#else
+static const char* kLPServerIdentifierKeyValue = "LP_SERVER_ID_KEY_VALUE is an unknown symbol";
+#endif
+
+int lp_tokenize_str (char *str, char c, char ***arr);
+
+int lp_tokenize_str (char *str, char c, char ***arr) {
+  int count = 1;
+  int token_len = 1;
+  int i = 0;
+  char *p;
+  char *t;
+
+  p = str;
+  while (*p != '\0') {
+    if (*p == c) {
+      count++;
+    }
+    p++;
+  }
+
+  *arr = (char**) malloc(sizeof(char*) * count);
+  if (*arr == NULL) { return -1; }
+
+  p = str;
+  while (*p != '\0') {
+    if (*p == c) {
+      (*arr)[i] = (char*) malloc( sizeof(char) * token_len );
+      if ((*arr)[i] == NULL) { return -1; }
+      token_len = 0;
+      i++;
+    }
+    p++;
+    token_len++;
+  }
+
+  (*arr)[i] = (char*) malloc( sizeof(char) * token_len );
+  if ((*arr)[i] == NULL) {  return -1; }
+
+  i = 0;
+  p = str;
+  t = ((*arr)[i]);
+  while (*p != '\0') {
+    if (*p != c && *p != '\0') {
+      *t = *p;
+      t++;
+    } else {
+      *t = '\0';
+      i++;
+      t = ((*arr)[i]);
+    }
+    p++;
+  }
+
+  return count;
+}
+
+int ___calabashserverinit();
+
+int ___lesspainfulserver = ___calabashserverinit();
+
+int ___calabashserverinit() {
```

Comment:
I discovered that depending on what Objective-C is included in this method, linking the framework will fail with:

```
"___gxx_personality_v0", referenced from: LPI.o
```

- indicating that `-lc++` linker flag is required.